lib/client.js

```diff
@@ -386,6 +386,34 @@ Client.prototype.delete = function (query, options, finalCallback) {
   return this._requestPost(this.UPDATE_PATH, bodyData, options, finalCallback);
 };
 
+/**
+ * Delete by id
+ * It will use the delete by id method:
+ * https://solr.apache.org/guide/8_11/uploading-data-with-index-handlers.html#delete-operations
+ *
+ * @param {string[]} ids - ids to delete
+ * @param {Object|Function} [options] - delete options
+ * @param {Function} finalCallback - (err, result)
+ *
+ * @returns {Undefined|Promise}
+ */
+Client.prototype.deleteById = function (ids, options, finalCallback) {
+  var bodyData;
+  if (arguments.length < 3 && _.isFunction(options)) {
+    finalCallback = options;
+    options = { commit: true }; // 'commit:true' option is default
+  }
+
+  if (Array.isArray(ids)) {
```

Comment:
.. and if its not an array...? Can you delete one item by id?

Reply:
ideally... I'd expect anyone using this pass in an array...

Reply:
So what happens if they don't and we send an empty or undefined body to solr?

Reply:
You may as well put the post inside this check and return false or something if its not an array